### PR TITLE
Add type hints to optimize.py.

### PR DESCRIPTION
**Pull request by bigbird555**
## Motivation

Please refer to the Issue #711 .

## Description of the changes

I added some type hints to lightgbm_tuner/optimize.py.

### CODE REVIEW

File: optuna/integration/lightgbm_tuner/optimize.py [hunk 1]
```diff
@@ -62,8 +62,9 @@
```
Comment (keisuke-umezawa) [inline optuna/integration/lightgbm_tuner/optimize.py:62]: Could you delete `if type_checking.TYPE_CHECKING:` in line 30?

Comment (bigbird555) [reply optuna/integration/lightgbm_tuner/optimize.py:61]: Okay.

**Comment by keisuke-umezawa** [comment]
@bigbird555 
Thank you!
Could you rebase your branch against current master branch to fix ci?

**Comment by bigbird555** [comment]
This PR does not still pass all the CI tests but....is it really okay?

**Comment by keisuke-umezawa** [comment]
@bigbird555 Sorry, you need to fix it to pass CI 🙇

**Pull request merged**